Added recipe for Pandemy version 1.0.0

antonlydell opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Once the PR is ready for review, please contact
`@conda-forge/staged-recipes` to have it merged by writing
<code>@</code>`conda-forge-admin, please ping team` in a new comment.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team][1] using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel][2] if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/pandemy```) and found some lint.

Here's what I've got...


For **recipes/pandemy**:

 * There are too few lines.  There should be one empty line at the end of the file.
 * ``requirements: host: setuptools>=42.0`` must contain a space between the name and the pin, i.e. ``setuptools >=42.0``
 * ``requirements: run: python>=3.8`` must contain a space between the name and the pin, i.e. ``python >=3.8``
 * ``requirements: run: pandas>=1.2.0`` must contain a space between the name and the pin, i.e. ``pandas >=1.2.0``
 * ``requirements: run: sqlalchemy>1.4.0`` must contain a space between the name and the pin, i.e. ``sqlalchemy >1.4.0``
 * If python is a host requirement, it should be a run requirement.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/pandemy```) and found some lint.

Here's what I've got...


For **recipes/pandemy**:

 * Non noarch packages should have python requirement without any version constraints.

antonlydell commented:
Dear conda-forge team,

I need some advice on my recipe.

Pandemy has the following requirements to run:
- python >=3.8
- pandas >=1.2.0
- sqlalchemy >=1.4.0

It further needs setuptools >=42 to build.

I get the linting issue: "Non noarch packages should have python requirement without any version constraints."
I want to set the run requirement to python >= 3.8. How do I do this?

Both pandas and SQLAlchemy are non noarch packages. Should I add noarch: python to the build section in meta.yaml?
Pandemy can run on any platform as long as pandas and SQLAlchemy are installed.

Thanks and Happy Easter
@conda-forge-admin please ping conda-forge/python

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/python and so here I am doing that.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/pandemy```) and found it was in an excellent condition.

antonlydell commented:
I hereby confirm that I (antonlydell) am willing to be become a maintainer of the recipe for Pandemy.

antonlydell commented:
Dear conda-forge team,

The recipe for Pandemy has passed the checks and is ready for review and merge.
@conda-forge/staged-recipes

Wishing you a Happy Easter,

Best regards
Anton Lydell

beenje commented on recipes/pandemy/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+{% set name = "Pandemy" %}
+{% set version = "1.0.0" %}
```
I noticed that version 1.1.0 is already available. Do you really want to publish 1.0.0 as conda package? Or should you bump to 1.1.0 this recipe?

antonlydell replied:
Hi beenje!

Yes I want to publish version 1.0.0 before version 1.1.0. I then want to go through the process of releasing a new version of the recipe (version 1.1.0) as practice exercise.

beenje commented on recipes/pandemy/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
```
The standard script command for Python packages is `script: {{ PYTHON }} -m pip install . -vv`.
`--no-deps` isn't needed as it is added automatically by conda-build.

No need to change. This is just a comment.

antonlydell replied:
Thanks for the feedback I will remove --no-deps in version 1.1.0.

beenje commented:
The recipe looks good to me. You found the answers to your questions yourself :-)
I just have one comment about the version.

antonlydell commented:
Sounds good!
Yes, after some more research about noarch: python I found that it made the most sense for Pandemy.
I want to publish version 1.0.0 before version 1.1.0 and then go through the process of releasing a new version of the recipe (version 1.1.0) as practice exercise.

antonlydell commented:
Dear beenje,

I have updated the recipe for Pandemy to version 1.1.0 (the most recent version) and the checks have passed.
I have also removed --no-deps from the pip install script as you suggested before.
Please review the changes.

Best regards
Anton Lydell
@conda-forge-admin, please ping team

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/staged-recipes and so here I am doing that.
